app/assets/javascripts/discourse/app/models/user.js

```diff
@@ -1133,6 +1139,68 @@ User.reopenClass(Singleton, {
   },
 });
 
+// user status tracking
+User.reopen(Evented, {
+  _clearStatusTimerId: null,
+
+  // always call stopTrackingStatus() when done with a user
+  trackStatus() {
+    this.addObserver("status", this, "_statusChanged");
+
+    if (this.isCurrent) {
+      this.appEvents.on("current-user-status:changed", (status) => {
```

Comment:
im not sure the off() will work correctly when passing an anonymous function to the on()